mcsema/Arch/ABI.cpp

```diff
@@ -455,14 +493,73 @@ static llvm::Function *WriteIntToMemFunc(uint64_t size_bytes) {
   }
 }
 
-llvm::Value *CallingConvention::LoadNextArgument(llvm::BasicBlock *block,
-                                                 llvm::Type *goal_type) {
+llvm::Value* InsertIntoVector(llvm::BasicBlock *block,
+                              llvm::Value *base_value,
+                              llvm::Value *reg_ptr,
+                              size_t count, size_t start=0) {
+
+  llvm::IRBuilder<> ir(block);
+  for (size_t i = 0; i < count; ++i) {
+    auto offset = ir.CreateGEP(reg_ptr, GetConstantInt(64, i));
+    auto load = ir.CreateLoad(offset);
+    base_value = ir.CreateInsertElement(base_value, load,
+        GetConstantInt(64, i + start));
+  }
+  return base_value;
+
+}
+
+// In llvm types that are expected to go into vector registers will
+// be of vector type, for example: <2 x float> will go into one %xmm.
+// External library must be compiled for the same architecture that
+// is passed to mcsema-lift, otherwise types may be wrong.
+llvm::Value *CallingConvention::LoadVectorArgument(
+    llvm::BasicBlock *block,
+    llvm::VectorType *goal_type) {
+
+  llvm::IRBuilder<> ir(block);
+  llvm::Value *base_value = llvm::Constant::getNullValue(goal_type);
+
+  auto under_type = goal_type->getElementType();
+  auto num_elements = goal_type->getNumElements();
+
+  size_t reg_size = GetVectorRegSize();
+
+  llvm::DataLayout dl(gModule);
+  auto element_size = dl.getTypeAllocSize(under_type);
+  size_t reg_element_capacity = reg_size / element_size;
+  int32_t remaining = static_cast<int32_t>(num_elements);
+
+  for ( unsigned i = 0; remaining > 0; ++i, remaining -= reg_size) {
+    auto reg_var_name = GetVarForNextArgument(goal_type);
+    LOG_IF(FATAL, !reg_var_name)
+      << "Could not find available vector register";
+
+    llvm::Value *dest_loc = remill::FindVarInFunction(block, reg_var_name);
+    dest_loc = ir.CreateBitCast(dest_loc,
```

Comment:
Move `dest_loc` to next line, or `llvm::..` to align with `dest_loc`, or don't split lines.